### DISCUSSION TITLE
etcd-v3.5.1 discovery failed","error":"couldn't find local

### DISCUSSION THREAD

**Discussion by sudheshpn** [etcd-io/etcd#16187] [Q&A] [answered]
Hi Experts,
                  We have upgraded etcd to v3.5.1 on one of our multimaster cluster. On 2 node the master node etcd works fine with similar configuration. On the 3rd etcd node we are seeing this error. Any clue what could be wrong. DNS name resolution to momv0316 works fine anon the host. Any clue what could be wrong?
 
 {"level":"fatal","ts":"2023-07-06T04:02:04.138Z","caller":"etcdmain/etcd.go:203","msg":"discovery failed","error":"couldn't find local name \"momv0316\" in the initial cluster configuration","stacktrace":"go.etcd.io/etcd/server/v3/etcdmain.startEtcdOrProxyV2\n\t/tmp/etcd-release-3.5.1/etcd/release/etcd/server/etcdmain/etcd.go:203\ngo.etcd.io/etcd/server/v3/etcdmain.Main\n\t/tmp/etcd-release-3.5.1/etcd/release/etcd/server/etcdmain/main.go:40\nmain.main\n\t/tmp/etcd-release-3.5.1/etcd/release/etcd/server/main.go:32\nruntime.main\n\t/home/remote/sbatsche/.gvm/gos/go1.16.3/src/runtime/proc.go:225"}

**Comment by sudheshpn** [comment]
Please find the unit file

[Unit]
Description=etcd

[Service]
Type=notify
ExecStart=/usr/local/bin/etcd \
  --name ${HOSTNAME} \                  --> Tried adding momv0316 
  --data-dir=/var/lib/etcd \
  --cert-file=/etc/kubernetes/pki/etcd/etcd.pem \
  --key-file=/etc/kubernetes/pki/etcd/etcd-key.pem \
  --peer-cert-file=/etc/kubernetes/pki/etcd/etcd.pem \
  --peer-key-file=/etc/kubernetes/pki/etcd/etcd-key.pem \
  --trusted-ca-file=/etc/kubernetes/pki/etcd/ca.pem \
  --peer-trusted-ca-file=/etc/kubernetes/pki/etcd/ca.pem \
  --peer-client-cert-auth \
  --client-cert-auth \
  --initial-advertise-peer-urls https://10.218.30.182:2380 \
  --listen-peer-urls https://10.218.30.182:2380 \
  --advertise-client-urls https://10.218.30.182:2379 \
  --listen-client-urls https://10.218.30.182:2379,https://127.0.0.1:2379 \
  --initial-cluster-token etcd-cluster-1 \
  --initial-cluster etcd1=https://10.218.30.178:2380,etcd2=https://10.218.30.181:2380,etcd3=https://10.218.30.182:2380 \
  --initial-cluster-state existing
Restart=on-failure
RestartSec=5

[Install]
WantedBy=multi-user.target

**Comment by jmhbnz** [comment, marked as answer]
This was resolved in #etcd slack https://kubernetes.slack.com/archives/C3HD8ARJ5/p1688621355178889?thread_ts=1688616862.519389&cid=C3HD8ARJ5 

> the `--name` parameter has to match a name that is in `--initial-cluster` list.
